lib/mapping.gi

```diff
@@ -1597,3 +1597,14 @@ function(f,t)
     SetRange(t,Range(f));
   fi;
 end);
+
+InstallMethod( String, 
+    "for an IdentityMapping", 
+    [ IsEndoGeneralMapping and IsOne ], 
+    function( id ) 
+    local str; 
+    str := String( Source( id ) ); 
+    str := Concatenation( "IdentityMapping( ", str, " )" ); 
+    ConvertToStringRep( str );
+    return str;
+end );
```

Comment:
I think it would be better to put this near the `ViewObj` and `PrintObj` methods, and also match their filters and coding style. That would be in `lib/mapprep.gi`, roughly around line 1343, and it also involves dynamic rank adjustment.

Reply:
Now moved to mapprep.gi as suggested, added the rank adjustment, and simplified the code.